api-analytics/service-worker.js

```diff
@@ -11,19 +11,30 @@ self.onactivate = function(event) {
   event.waitUntil(self.clients.claim());
 };
 
-// Fetch is as simply as log the request and passthrough.
-// Water clear thanks to the promise syntax!
 self.onfetch = function(event) {
-  event.respondWith(log(event.request).then(fetch));
+  event.respondWith(
+    // Log the request ...
+    log(event.request)
+    // .. and then actually perform it.
+    .then(fetch)
+  );
 };
 
 // Post basic information of the request to a backend for historical purposes.
 function log(request) {
-  var returnRequest = function() { return Promise.resolve(request); };
-  var data = { method: request.method, url: request.url };
+  var returnRequest = function() {
+    return request;
```

Comment:
Of course, it can appear in the middle of the chain due to reason 1. The advantage of the other form is that it can appear in **both** places without altering. The good thing with promises, is that they are value abstractions. Wrapping operations in promise-returning functions allows you to hide synchronous / asynchronous details, working in an homogeneous way. The same can not be said for synchronous programming but... as said, not blocking on this.